Sources/Confidence/Confidence.swift

```diff
@@ -3,47 +3,127 @@ import Combine
+        flagApplier: FlagApplier,
+        remoteFlagResolver: ConfidenceResolveClient,
+        storage: Storage,
         context: ConfidenceStruct = [:],
         parent: ConfidenceEventSender? = nil,
         visitorId: String? = nil
     ) {
         self.eventSenderEngine = eventSenderEngine
         self.clientSecret = clientSecret
+        self.timeout = timeout
         self.region = region
-        self.initializationStrategy = initializationStrategy
-        self.contextFlow.value = context
+        self.storage = storage
+        self.contextSubject.value = context
         self.parent = parent
+        self.storage = storage
+        self.flagApplier = flagApplier
+        self.remoteFlagResolver = remoteFlagResolver
         if let visitorId {
             putContext(context: ["visitor_id": ConfidenceValue.init(string: visitorId)])
         }
+
+        contextChanges().sink { [weak self] context in
+            guard let self = self else {
+                return
+            }
+            self.currentFetchTask?.cancel()
+            self.currentFetchTask = Task {
+                do {
+                    let context = self.getContext()
+                    try await self.fetchAndActivate()
+                    self.contextReconciliatedChanges.send(context.hash())
+                } catch {
+                }
+            }
+        }
+        .store(in: &cancellables)
     }
 
-    public func track(eventName: String, message: ConfidenceStruct) {
-        eventSenderEngine.emit(eventName: eventName, message: message, context: getContext())
+    public func activate() throws {
+        let savedFlags = try storage.load(defaultValue: FlagResolution.EMPTY)
+        self.cache = savedFlags
+    }
+
+    public func fetchAndActivate() async throws {
+        try await internalFetch()
+        try activate()
+    }
+
+    func internalFetch() async throws {
+        let context = getContext()
+        let resolvedFlags = try await remoteFlagResolver.resolve(ctx: context)
+        let resolution = FlagResolution(
+            context: context,
+            flags: resolvedFlags.resolvedValues,
+            resolveToken: resolvedFlags.resolveToken ?? ""
+        )
+        try storage.save(data: resolution)
+    }
+
+    public func asyncFetch() {
+        Task {
+            try await internalFetch()
+        }
+    }
+
+    public func getEvaluation<T>(key: String, defaultValue: T) throws -> Evaluation<T> {
+        try self.cache.evaluate(
+            flagName: key,
+            defaultValue: defaultValue,
+            context: getContext(),
+            flagApplier: flagApplier
+        )
+    }
+
+    public func getValue<T>(key: String, defaultValue: T) -> T {
+        do {
+            return try getEvaluation(key: key, defaultValue: defaultValue).value
+        } catch {
+            return defaultValue
+        }
+    }
+
+    func isStorageEmpty() -> Bool {
+        return false
```

Comment:
A WIP?

Reply:
we should use storage here. let's update it right away, i thought i already added it. it seems i didn't :)